clients/cloud/ruby/c_cloud.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/ruby
+#
+# Copyright 2018 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# =============================================================================
+#
+# Helper module
+#
+# =============================================================================
+
+require 'optparse'
+require 'kafka'
+
+#
+# Helper class for working with Confluent Cloud
+#
+class CCloud
+  attr_reader :kafka
+
+  def initialize
+    # Parse arguments, load the CCloud config and initialize the Kafka client
+    @args = parse_args!
+    @config = load_config(@args[:config])
+    @kafka = Kafka.new(
+      seed_brokers: @config[:'bootstrap.servers'],
+      sasl_plain_username: @config[:'sasl.username'],
+      sasl_plain_password: @config[:'sasl.password'],
+      ssl_ca_certs_from_system: true
+    )
+  end
+
+  def topic
+    @args[:topic]
+  end
+
+  def consumer
+    return @consumer unless @consumer.nil?
+
+    # Consumers with the same group id will form a Consumer Group together.
+    @consumer = @kafka.consumer(group_id: 'ruby_example_group_1')
+
+    # Stop the consumer when the SIGTERM signal is sent to the process.
```

Comment:
Since this runs in the foreground I'd suggest SIGINT (ctrl-c) instead (or both)

Reply:
This actually catches all and is a left-over comment - so I'll just remove it